extractor/src/test/java/com/example/extractor/MainsLauncherTest.java

```diff
@@ -139,8 +145,81 @@ void testGetTemplateFilePathForInputFile() throws Exception {
         assertThat(templatePath).endsWith("config-template-treewalker.xml");
     }
 
+    /**
+     * Helper method to load a file's content into a string.
+     *
+     * @param filePath The path to the file.
+     * @return The file's content as a string.
+     * @throws IOException If an I/O error occurs.
+     */
     private String loadToString(final String filePath) throws IOException {
         final byte[] encoded = Files.readAllBytes(Paths.get(filePath));
         return new String(encoded, StandardCharsets.UTF_8);
     }
+
+    /**
+     * Tests the external config files functionality.
+     */
+    @Test
+    void testMainWithExternalConfigFiles(@TempDir final Path tempDir) throws Exception {
+        // Create the 'config' directory inside the tempDir
+        final Path configDir = tempDir.resolve("config");
+        Files.createDirectories(configDir);
+
+        // Write the java.header file inside the 'config' directory
+        final String headerFileContent = """
+        ^// Copyright \\(C\\) (\\d\\d\\d\\d -)? 2004 MyCompany$
+        ^// All rights reserved$
+            """;
```

Comment:
please make it as static file in repo, that you will use during test execution.
same for config file.